### PR TITLE
[#55 and #56] Added Logout Functionality with Session Management and Implemented NavBar with Full Feature Set

### PR DESCRIPTION
**Pull request by saksham1919**
Authors: @Banna150149 @saksham1919 

Built a Navigation Bar for the AutoGrader-Frontend application.

Implemented Logout feature with session management.

## Description
Created a NavBar for the AutoGrader-Frontend application with routes to the Course Dashboard, Assignments, Manage Users, and Logout functionality.

Implemented session-based Logout functionality, ensuring that only authenticated users can access the application pages across all routes.

## How has this been tested?
Completed RSpec and Cucumber testing with full coverage of the newly added code.
Updated existing tests for all application controllers to incorporate session management and logout functionality, ensuring successful execution without any failures.

## Screenshots (if appropriate):
Login Screen:
image

Assignments View for a Logged in user:
image

SimpleCov Coverage report-
image

## Checklist:

- [*] Zero `rubocop` violations
- [*] My code receives an "A" when `rubycritic` is run
- [*] >90% test coverage of my changes
- [*] All new and existing tests passed

### CODE REVIEW

File: config/local_env.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 GITHUB_TEMPLATE_REPO_URL: 'philipritchey/autograded-assignment-template'
 GITHUB_COURSE_ORGANIZATION: 'AutograderFrontend'
 ASSIGNMENTS_BASE_PATH: 'assignment-repos/'
+GITHUB_CLIENT_ID: "Ov23liMDAgFqwvCIF4kD"
```
Comment (maxsmith271346) [inline config/local_env.yml:4]: We should not be pushing our client secret.

**Comment by riddhighate-tamu** [comment]
Need to fix a few things with navbar:
1. The navbar should not be visible on the login page.
2. I think you have displayed the navbar inside the div, which is incorrect, it should be on the top of the page. Something like in the image given below:

Screenshot 2024-10-18 at 12 25 09 PM

File: config/local_env.yml (hunk 1 above)
Comment (saksham1919) [reply config/local_env.yml:4]: Right, fixed that.

**Comment by maxsmith271346** [comment]
@Banna150149 @saksham1919 In the future, please limit a PR to a single feature (logout function in one, navbar in another)

**Comment by saksham1919** [comment]
> @Banna150149 @saksham1919 In the future, please limit a PR to a single feature (logout function in one, navbar in another)

Thanks for the feedback. The logout functionality and the navbar were closely linked in this case, as the logout option is provided within the navbar itself. However, I completely agree that PRs should ideally be focused on individual features, and we all should ensure to keep them more specific moving forward.

**Comment by saksham1919** [comment]
> Need to fix a few things with navbar:
> 
> 1. The navbar should not be visible on the login page.
> 2. I think you have displayed the navbar inside the div, which is incorrect, it should be on the top of the page. Something like in the image given below:
> 
> Screenshot 2024-10-18 at 12 25 09 PM


Yes, thank you for the feedback. As discussed, the NavBar is still present, but we have modified it to exclude links to the application pages, displaying only the app name. I’ve attached the updated screenshots with the revised views for your review.

**Review by riddhighate-tamu — APPROVED** [review]

**Pull request merged**